theano/gpuarray/dnn.py

```diff
@@ -2919,3 +2920,72 @@ def local_gpua_softmax_dnn_grad(op, ctx_name, inputs, outputs):
     out = GpuDnnSoftmaxGrad('accurate', 'instance')(
         gpu_contiguous(ins[0]), gpu_contiguous(ins[1]))
     return [out.dimshuffle(0, 2)]
+
+
+class GpuRoIPool(RoIPoolOp, COp):
+
+    __props__ = ('spatial_scale', 'pooled_h', 'pooled_w')
+    func_file = "./roi_pool.c"
+    func_name = "APPLY_SPECIFIC(GPUForward)"
+
+    def __init__(self, pooled_h, pooled_w, spatial_scale):
+        super(GpuRoIPool, self).__init__(pooled_h, pooled_w, spatial_scale)
+        self.pooled_h = pooled_h
+        self.pooled_w = pooled_w
+        self.spatial_scale = spatial_scale
+
+    def make_node(self, feature_maps, roi):
+        ctx_name = infer_context_name(feature_maps, roi)
+        feature_maps = as_gpuarray_variable(feature_maps, ctx_name)
+        roi_tuples = as_gpuarray_variable(roi, ctx_name)
```

Comment:
That is the right way to infer the context. So I don't understand the problem you have here.